File: producer/api/models/response.js
const mongoose = require("mongoose");

const responseSchema = new mongoose.Schema({
    _id: mongoose.Schema.Types.ObjectId,
    form: {type: mongoose.Schema.Types.ObjectId, ref: "Form",required: true},
    user: {type:mongoose.Schema.Types.ObjectId},
    submittedAt: { type: Date, default: Date.now }
})

module.exports = mongoose.model("Response", responseSchema);

File: go-producer/models/response.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type ResponseSchema struct {
	Id          primitive.ObjectID `bson:"_id,omitempty"`
	Form        primitive.ObjectID `bson:"form,omitempty" validate:"required"`
	User        primitive.ObjectID `bson:"user,omitempty" validate:"required"`
	SubmittedAt primitive.DateTime `bson:"submittedAt,omitempty"` // Default needs to be date.now()
}
